pvr: fix caching on stalled streams

pipelka opened the pull request:
Sometimes it happens that the TV livestream stalls. This may happen when the signal on the receiver is lost or the TCP/IP connection to the server is lost.

Currently its impossible to continue streaming when the stream is recovered because there are no more packets requested via DemuxRead. You need to press "Stop" and reselect the channel to start streaming again.

This patch fixes the problem (at least for me).

pipelka committed:
pvr: fix caching on stalled streams

FernetMenta commented:
see also #265

FernetMenta commented:
Ok, player stops reading demuxer. That's valuable information. The only case I see when this can happen is:

```
 if ((!m_dvdPlayerAudio.AcceptsData() && m_CurrentAudio.id >= 0) ||
        (!m_dvdPlayerVideo.AcceptsData() && m_CurrentVideo.id >= 0))
    {
      Sleep(10);
      continue;
    }
```

Player is paused and one queue is full. Would you be able to verify this? On my system this problem occurs not very often.

pipelka commented:
Ok. I added the message `sleeping. queue full.` for that case.

Here is the result:

<pre>
15:27:00 T:140365144209152   DEBUG: AddOnLog: VDR XVDR Client: DemuxRead
15:27:00 T:140365256328960   DEBUG: AddOnLog: VDR XVDR Client: SendPing
15:27:04 T:140365144209152   DEBUG: AddOnLog: VDR XVDR Client: DemuxRead
15:27:08 T:140365256328960   DEBUG: Previous line repeats 2 times.
15:27:08 T:140365256328960   DEBUG: AddOnLog: VDR XVDR Client: SendPing
15:27:10 T:140365144209152   DEBUG: AddOnLog: VDR XVDR Client: DemuxRead
15:27:16 T:140365256328960   DEBUG: Previous line repeats 3 times.
15:27:16 T:140365256328960   DEBUG: AddOnLog: VDR XVDR Client: SendPing
15:27:18 T:140365144209152   DEBUG: CAddonCallbacksAddon - QueueNotification - VDR XVDR Client - Info Message : 'TV Signal restored'
15:27:18 T:140365144209152   DEBUG: AddOnLog: VDR XVDR Client: changing to channel 47
15:27:18 T:140365717665728   DEBUG: ------ Window Init (DialogKaiToast.xml) ------
15:27:18 T:140365717665728   DEBUG: Alloc resources: 0.28ms (0.27 ms skin load)
15:27:19 T:140365144209152   DEBUG: AddOnLog: VDR XVDR Client: DemuxRead
15:27:19 T:140365144209152   DEBUG: AddOnLog: VDR XVDR Client: DemuxReset
15:27:19 T:140365144209152   DEBUG: CDVDDemuxPVRClient::RequestStreams(): added stream 0:301 with codec_id 2
15:27:19 T:140365144209152   DEBUG: CDVDDemuxPVRClient::RequestStreams(): added stream 1:303 with codec_id 86016
15:27:19 T:140365144209152   DEBUG: AddOnLog: VDR XVDR Client: DemuxRead
15:27:19 T:140365144209152   DEBUG: CDVDDemuxPVRClient::UpdateStreams(): update stream 0:301 with codec_id 2
15:27:19 T:140365144209152   DEBUG: CDVDDemuxPVRClient::UpdateStreams(): update stream 1:303 with codec_id 86016
15:27:19 T:140365144209152   DEBUG: AddOnLog: VDR XVDR Client: DemuxRead
15:27:19 T:140365144209152 WARNING: CDVDPlayer::CheckContinuity - resync forward :2, prev:56191869411.000000, curr:56236549411.000000, diff:44680000.000000
15:27:19 T:140365144209152   DEBUG: Process - sleeping. queue full.
15:27:22 T:140365717665728   DEBUG: Previous line repeats 226 times.
15:27:22 T:140365717665728   DEBUG: ------ Window Deinit (DialogKaiToast.xml) ------
15:27:22 T:140365144209152   DEBUG: Process - sleeping. queue full.
15:27:24 T:140365256328960   DEBUG: Previous line repeats 250 times.
15:27:24 T:140365256328960   DEBUG: AddOnLog: VDR XVDR Client: SendPing
15:27:24 T:140365144209152   DEBUG: Process - sleeping. queue full.
15:27:32 T:140365256328960   DEBUG: Previous line repeats 776 times.
15:27:32 T:140365256328960   DEBUG: AddOnLog: VDR XVDR Client: SendPing
15:27:32 T:140365144209152   DEBUG: Process - sleeping. queue full.
15:27:40 T:140365256328960   DEBUG: Previous line repeats 775 times.
15:27:40 T:140365256328960   DEBUG: AddOnLog: VDR XVDR Client: SendPing
15:27:40 T:140365144209152   DEBUG: Process - sleeping. queue full.
15:27:48 T:140365256328960   DEBUG: Previous line repeats 775 times.
15:27:48 T:140365256328960   DEBUG: AddOnLog: VDR XVDR Client: SendPing
15:27:48 T:140365144209152   DEBUG: Process - sleeping. queue full.
15:27:56 T:140365256328960   DEBUG: Previous line repeats 777 times.
15:27:56 T:140365256328960   DEBUG: AddOnLog: VDR XVDR Client: SendPing
15:27:56 T:140365144209152   DEBUG: Process - sleeping. queue full.
15:28:00 T:140365389498112   DEBUG: Previous line repeats 374 times.
15:28:00 T:140365389498112   DEBUG: Thread Jobworker start, auto delete: 1
15:28:00 T:140365144209152   DEBUG: Process - sleeping. queue full.
15:28:04 T:140365256328960   DEBUG: Previous line repeats 399 times.
15:28:04 T:140365256328960   DEBUG: AddOnLog: VDR XVDR Client: SendPing
15:28:04 T:140365144209152   DEBUG: Process - sleeping. queue full.
15:28:12 T:140365256328960   DEBUG: Previous line repeats 773 times.
15:28:12 T:140365256328960   DEBUG: AddOnLog: VDR XVDR Client: SendPing
15:28:12 T:140365144209152   DEBUG: Process - sleeping. queue full.
</pre>


Yes. Thats exactly what happens.

FernetMenta commented:
Great and thanks a lot for tracking this down! I will consult elupus on what's the best strategy to deal with this case.
